tests/cpp/test_matmul_scheduler.cpp

```diff
@@ -2832,6 +2832,105 @@ TEST_F(MatmulSchedulerTest, DISABLED_RequireExternalPlugin) {
   MatmulParams params;
 }
 
+class AllocationDomainTest
+    : public NVFuserFixtureParamTest<std::tuple<bool, bool>> {
+ protected:
+  // Allocation order set by the pass breaks matmul tests
+  // see issue https://github.com/NVIDIA/Fuser/issues/1810
+  AllocationDomainTest() : optimization_guard_(false) {
+    DisableOptionsGuard::getCurOptions().set(DisableOption::MatmulExprEval);
+  }
+
+ private:
+  preseg_passes::OptimizationPassGuard<preseg_passes::AllocationDomainPass>
+      optimization_guard_;
+
+  DisableOptionsGuard option_guard_;
+};
+
+// This tests fusions where inputs to a Matmul will have the root domains
+// [M, K] and [K, N], and all possible combinations of allocation domains.
+// Please note that inpout in B is transposed prior to creating a Mma op.
+TEST_P(AllocationDomainTest, BasicMatmul) {
+  NVFUSER_TEST_CUDA_ARCH_RANGE_GUARD(7, 5, 9, 0);
+  bool a_has_allocation = std::get<0>(GetParam());
+  bool b_has_allocation = std::get<0>(GetParam());
+
+  auto setupFusion = [](Fusion* fusion, TensorView* tv0, TensorView* tv1) {
+    fusion->addInput(tv0);
+    fusion->addInput(tv1);
+
+    // This has rfactor: {N, K}
+    auto tv1t = transpose(tv1);
+
+    // Propagate the allocation domain of B accross the transpose.
+    if (tv1->hasAllocation()) {
+      tv1t->setAllocationDomain({tv1t->axis(0), tv1t->axis(1)}, true);
+    }
```

Comment:
I think the scheduler should actually be the one setting the allocation domain for this tensor. It would set the allocation domain on the register tensor that is the argument to the MmaOp since that is where the TN requirement comes from, then propagate allocation domain back to the register tensor that gets loaded from smem (i.e. the ldmatrix output).